app/javascript/src/Dashboard/components/PotentialUpgradesWidget.jsx

```diff
@@ -0,0 +1,64 @@
+// @flow
+
+import React from 'react'
+
+import { createReactWrapper } from 'utilities'
+import pluralize from 'pluralize'
+
+import 'Dashboard/styles/dashboard.scss'
+
+type Props = {
+  violations: Array<{
+    id: number,
+    account_id: number,
+    account_name: string,
+    alerts_count: number,
+    url: string
+  }>,
+  incorrectSetUp: boolean,
+  links: {
+    adminServiceApplicationPlans: string,
+    settingsAdminService: string
+  }
+}
+
+const PotentialUpgradesWidget = ({ violations, incorrectSetUp, links }: Props): React.Node => {
+  return (
+    <article className='DashboardWidget' id='dashboard-widget-potential_upgrades'>
+      <h1 className='DashboardWidget-title'>
+        Potential Upgrades
+      </h1>
+      <span className='DashboardWidget-intro'>
+        Accounts that hit their Usage Limits in the last 30 days
+      </span>
+
+      {
+        violations.some((violation) => violation)
+          ? <ol className='DashboardWidgetList'>
+            {
+              violations.map(violation =>
+                <li className='DashboardWidgetList-item' key={violation.account_id}>
+                  <a href={violation.url} className='DashboardWidgetList-link'>
+                    {violation.account_name} has {violation.alerts_count} { pluralize('Usage Limit alert', violation.alerts_count) }
+                  </a>
+                </li>
+            )}
+          </ol>
+          : (incorrectSetUp)
```

Comment:
Seems to me a guard condition... why check this only if violations is empty? I assume it was like that before but we could improve it, maybe.
IMO `violations` should be `nil` if not setup correctly and the guard could be something like.
```jsx
if (incorrectSetUp) {
  return (
    <>
      <p> In order to show... </p>
      <p> ... </p>
    </>
  )
}

return ( ... )

Reply:
I changed the order in b89bf8e but, since there's an elsif and an else, I didn't manage to scape the ternary.
Should I create a function that receives **all** arguments, check what should be returned and then call it in the JSX? Is there a better way to do it? 😬